src/main/java/io/pravega/connectors/flink/FlinkPravegaReader.java

```diff
@@ -488,18 +574,40 @@ protected ReaderGroupManager createReaderGroupManager() {
          * Sets the deserialization schema.
          *
          * @param deserializationSchema The deserialization schema
+         * @return Builder instance.
          */
         public Builder<T> withDeserializationSchema(DeserializationSchema<T> deserializationSchema) {
             this.deserializationSchema = deserializationSchema;
             return builder();
         }
 
+        /**
+         * Sets the timestamp and watermark assigner.
+         *
+         * @param assignerWithTimeWindows The timestamp and watermark assigner.
+         * @return Builder instance.
+         */
+        protected Builder<T> withTimestampAndWatermark(AssignerWithTimeWindows<T> assignerWithTimeWindows) {
+            try {
+                ClosureCleaner.clean(assignerWithTimeWindows, true);
+                this.assignerWithTimeWindows = new SerializedValue<>(assignerWithTimeWindows);
+            } catch (IOException e) {
+                throw new IllegalArgumentException("The given assigner is not serializable", e);
+            }
+            return this;
+        }
+
         @Override
         protected DeserializationSchema<T> getDeserializationSchema() {
             Preconditions.checkState(deserializationSchema != null, "Deserialization schema must not be null.");
             return deserializationSchema;
         }
 
+        @Override
+        protected SerializedValue<AssignerWithTimeWindows<T>> getAssignerWithTimeWindows() {
```

Comment:
I can accept this, but it might be confused with Flink's `org.apache.flink.streaming.api.functions.TimestampAssigner` which only contains the `extractTimestamp` method. Our `AssignerWithTimeWindows` extends that, so it has not only this method but also `getWatermark(TimeWindow)` method.